irksome/stage.py

```diff
@@ -361,14 +418,70 @@ def __init__(self, F, butcher_tableau, t, dt, u0, bcs=None,
         self.update_problem = NonlinearVariationalProblem(
             Fupdate, unew, update_bcs)
 
+        if bounds is not None and bounds[0] == "time_level":
+            assert update_solver_parameters is not None, \
+                "update_solver_parameters required for bounds-constrained update"
+
         self.update_solver = NonlinearVariationalSolver(
             self.update_problem,
             solver_parameters=update_solver_parameters)
 
-        if stiff_acc_huh:
-            self._update = self._update_stiff_acc
-        else:
-            self._update = self._update_general
+        if bounds is not None:
+            V = u0.function_space()
+            bounds_type, lower, upper = bounds
+            lower_func = Function(V)
+            upper_func = Function(V)
+            if not isiterable(lower):
+                lower = (lower,)
+            if not isiterable(upper):
+                upper = (upper,)
+            for l, lf in zip(lower, lower_func.subfunctions):
+                if l is not None:
+                    lf.assign(l)
```

Comment:
Are these `assign`s unavoidable?  Not that we've made much progress on being adjoint-able, but this seems bad from that perspective...

Reply:
I thought that `dat` bad, `assign` good?